### ISSUE TITLE
How to add support for new datasources?

### ISSUE THREAD

**Issue by raakasf** [K-Phoen/grabana#184]
Hello,

I really like the dashboard as code concept that Grabana has implemented. I work on a community driven [druid datasource](https://github.com/grafadruid/druid-grafana). I was wondering how to go about adding support for a new datasource?

-R

**Comment by K-Phoen** [comment]
Hey!

Glad you like it :)

Supporting new datasources requires a few steps:
* add support to create the datasource as code → checkout the [datasource/ folder for examples](https://github.com/K-Phoen/grabana/tree/master/datasource). Every datasource in there implements [the same interface](https://github.com/K-Phoen/grabana/blob/master/datasource/datasource.go), which allows the client to create/update datasources definitions
* define a "target" type for the druid datasource, [as we do for other ones](https://github.com/K-Phoen/grabana/tree/master/target). This allows panels to query the datasource
* add an option to define a Druid query/target for each panel that should be able to. [Example for the "graph" panel](https://github.com/K-Phoen/grabana/blob/f6968d71b937168b9192ce5a908d855eaf183378/graph/graph.go#L138-L156)
* repeat the process for alerts: https://github.com/K-Phoen/grabana/tree/master/alert/queries (both the query builder and option to use it in different panels)

Is adding support for druid something you would be willing to contribute yourself? I can provide some support if needed :)

**Comment by raakasf** [comment]
Yes, I am willing to contribute! I'll reach out to you when I start working on this! Thanks.
